expat/configure.ac

```diff
@@ -103,7 +103,7 @@ AC_SUBST(LIBREVISION)
 AC_SUBST(LIBAGE)
 
 AC_LANG([C])
-AC_PROG_CC_C99
+AC_PROG_CC
```

Comment:
This seems to be no loss in current functionality (because the C99 compiler is not currently enforced by mere use of `AC_PROG_CC_C99`) but it's a loss in intention/information and we actually want to require a C99 compiler before `make` comes into play. So if this is touched, we should probably compare `$ac_cv_prog_cc_c99` to `no` here using `AS_IF` and `AC_MSG_ERROR`. Either that or a revert of this hunk please.

Reply:
PS: This seems to work well for a pre-C99 compiler when testing:
```console
# cat c89.sh 
#! /usr/bin/env bash
set -x
exec gcc "$@" -std=c89
```

Reply:
PPS: `master` now has #722 merged so a rebase onto latest `master` will give this pull request a realistic chance for a green CI.

Reply:
@AtariDreams how would you like to proceed?